src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -51,18 +51,21 @@ public RuntimeMetrics(RuntimeMetricsOptions options)
 
             if (options.IsGcEnabled)
             {
-                // TODO: Almost all the ObservableGauge should be ObservableUpDownCounter (except for CPU utilization).
-                // Replace them once ObservableUpDownCounter is available.
-                this.meter.CreateObservableGauge($"{metricPrefix}gc.count", () => GetGarbageCollectionCounts(), description: "GC Count for all generations.");
+                this.meter.CreateObservableCounter($"{metricPrefix}gc.count", () => GetGarbageCollectionCounts(), description: "Number of times garbage collection has occurred since process start.");
 
 #if NETCOREAPP3_1_OR_GREATER
-                this.meter.CreateObservableCounter($"{metricPrefix}gc.allocated.bytes", () => GC.GetTotalAllocatedBytes(), "By", "Allocation Rate.");
+                this.meter.CreateObservableCounter($"{metricPrefix}gc.allocated", () => GC.GetTotalAllocatedBytes(), unit: "By", description: "Count of the bytes allocated since the process start, this does not include any native allocations.");
```

Comment:
```suggestion
                this.meter.CreateObservableCounter($"{metricPrefix}gc.allocated", () => GC.GetTotalAllocatedBytes(), unit: "By", description: "Count of the bytes allocated on the managed GC heap since the process start. .NET objects are allocated from this heap. Object allocations from unmanaged languages such as C/C++ do not use this heap.");
```
Any downside you are aware of to having the descriptions be somewhat verbose? I'm suspicious that if folks didn't have an intuitive understanding of what the "gc" means in the metric name they may not know what a 'native allocation' is either.

Reply:
Given metrics are only reported at low frequency (e.g. every few seconds or even minutes), I guess the only limit is the UI (e.g. I imagine the description to be a tip/popup or something rendered in the UI). So being verbose should be okay in my opinion.